UI/FleetWnd.cpp

```diff
@@ -3550,11 +3561,38 @@ void FleetWnd::FleetRightClicked(GG::ListBox::iterator it, const GG::Pt& pt, con
                     boost::dynamic_pointer_cast<GiveObjectToEmpireOrder>(it->second))
                 {
                     if (order->ObjectID() == fleet->ID()) {
-                        HumanClientApp::GetApp()->Orders().RecindOrder(it->first);
+                        HumanClientApp::GetApp()->Orders().RescindOrder(it->first);
                         // could break here, but won't to ensure there are no problems with doubled orders
                     }
                 }
             }
+            break;
+        }
+
+        case 13: { // Remove visibility information for this fleet from
+                   // the empire's visibility table.
+            // Server changes for permanent effect
+            // Tell the server to change what the empire wants to know
+            // in future so that the server doesn't keep resending this
+            // fleet information.
+            HumanClientApp::GetApp()->Orders().IssueOrder(
+                OrderPtr(new ForgetOrder(client_empire_id, fleet->ID()))
+            );
+            // Client changes for immediate effect
+            // Force the client to change immediately.
+            GetUniverse().ForgetKnownObject(ALL_EMPIRES, fleet->ID());
+
+            // Force a redraw
+
+            // Just signaling StateChanged doesn't work because MapWnd
+            // only resets/deletes paths to non-existent objects on the
+            // turn updates and not when handling StateChanged.
+            // fleet->StateChangedSignal();
+
+            //So brute force.
+            ClientUI::GetClientUI()->GetMapWnd()->MidTurnUpdate();
```

Comment:
I think MidTurnUpdate is doing a lot more than is needed at this point, including recalculating the system graph connectivity and refreshing all the system icon and fleet icons. Should be possible to something more specific, like

````
            this->Refresh();
            ClientUI::GetClientUI()->GetMapWnd()->RefreshFleetButtons();
````

MapWnd::RefreshFleetButtons is private though, so would need to be moved to the public section of the MapWnd definition. It's also refreshing all the fleet buttons, which is probably not necessary, but fixing that would be a bit more involved.